receiver/couchbasereceiver/models.go

```diff
@@ -0,0 +1,86 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package couchbasereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchbasereceiver"
+
+// Cluster response models
+
+type clusterResponse struct {
```

Comment:
The concern I always have with unmarshaling responses into structs is that we may lose the ability to differentiate between missing data points and actual zero values. Do you know if we can count on the couchbase response to always be complete?

Reply:
[This](https://docs.couchbase.com/server/current/rest-api/rest-bucket-stats.html#response-2) says `Depending on the level of bucket activity, there may be more detail for each field or less.`, which isn't perfectly clear, but certainly begs the question as to whether or not fields will be omitted due to lack of relevant activity.

Reply:
If we switch to  pointers for data point values we won't lose that differentiation which is something I can do for the few metrics in the structs. The bulk of metrics will be in a `map[string]interface{}` on the `bucketStats` struct. 

That being said the ones I did specifically add fields for should be safe. I can add pointer fields for those just so we can null check them (would be the same as an `val, ok := map[key]` type check but no type casting of the value).

Reply:
Up to you - just need to make sure we are not generating phantom data points.